tasks/main-tang.yml

```diff
@@ -1,81 +1,51 @@
 ---
 - name: Ensure tang is installed
   package:
-    name: "{{ __nbde_server_package_list }}"
+    name: "{{ __nbde_server_packages }}"
     state: present
 
 
-- name: Ensure tang is enabled and started
-  when: state is not defined or state != "stopped"
-  service:
-    name: "{{ __nbde_server_service }}"
-    enabled: yes
-    state: started
-
-
-- name: Ensure tang is enabled but stopped
-  when: state is defined and state == "stopped"
-  service:
-    name: "{{ __nbde_server_service }}"
-    enabled: yes
-    state: stopped
-
-
 - name: Ensure keys are rotated
   when: nbde_server_rotate_keys
   nbde_server_tang:
-    state: keys-rotated
+    action: rotate-keys
     keygen: "{{ __nbde_server_keygen }}"
     keydir: "{{ __nbde_server_keydir }}"
+  notify: nbde_server tang cache update
 
 
 - name: Ensure we have keys
   when: not nbde_server_rotate_keys
   nbde_server_tang:
-    state: keys-created
+    action: create-keys
     keygen: "{{ __nbde_server_keygen }}"
     keydir: "{{ __nbde_server_keydir }}"
+  notify: nbde_server tang cache update
 
 
-- name: Ensure we fetch the server keys
-  when: nbde_server_fetch_keys
-  synchronize:
-    mode: pull
-    src: "{{ __nbde_server_keydir }}/"
-    dest: "{{ nbde_server_keys_dir }}/{{ inventory_hostname }}"
-    delete: yes
-    rsync_opts:
-      - "--chmod=D2755,F644"
+- name: Perform key management (fetch/deploy) tasks
+  when: nbde_server_fetch_keys or nbde_server_deploy_keys
+  include: tang-key-management.yml
+  notify: nbde_server tang cache update
 
 
-- name: Ensure we fetch the created keys for deploying
-  when: nbde_server_deploy_keys
-  run_once: yes
-  synchronize:
-    mode: pull
-    src: "{{ __nbde_server_keydir }}/"
-    dest: "{{ nbde_server_keys_dir }}"
-    delete: yes
-    rsync_opts:
-      - "--chmod=D2755,F644"
-
-
-- name: Ensure we copy specific keys to tang
-  when: nbde_server_deploy_keys
-  check_mode: no
-  synchronize:
-    mode: push
-    src: "{{ nbde_server_keys_dir }}"
-    dest: "{{ __nbde_server_keys_to_deploy_dir }}"
+- name: Ensure tang is enabled and started
+  when: state is not defined or state != "stopped"
+  service:
+    name: "{{ __nbde_server_service }}"
+    enabled: yes
+    state: started
+  notify: nbde_server tang cache update
 
 
-- name: Ensure we deploy the specific keys to tang
-  when: nbde_server_deploy_keys
-  nbde_server_tang:
-    state: keys-deployed
-    keydir: "{{ __nbde_server_keydir }}"
-    keys_to_deploy_dir: "{{ nbde_server_keys_dir }}"
-    base_keys_to_deploy_dir: "{{ __nbde_server_keys_to_deploy_dir }}"
+- name: Ensure tang is enabled but stopped
+  when: state is defined and state == "stopped"
+  service:
+    name: "{{ __nbde_server_service }}"
+    enabled: yes
+    state: stopped
```

Comment:
I don't understand what this is doing?

Reply:
This should be stopping the service

Reply:
It sounds like every other service that you can use e.g. `systemctl stop NAME` to manage.  In which case, I think you can just ensure the service is enabled and started e.g. https://github.com/linux-system-roles/kernel_settings/blob/master/tasks/main.yml#L16
```yaml
- name: Ensure required services are enabled and started
  service:
    name: "{{ item }}"
    state: started
    enabled: yes
  loop: "{{ __kernel_settings_services }}"
```